[CI] MixedClusterClientYamlTestSuiteIT test {p0=search.vectors/40_knn_search/Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn} failing

**Build Scans:**
- [elasticsearch-intake #41492 / 9.3.6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/5jiicprzvjht2)
- [elasticsearch-pull-request #154008 / 9.3.6_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/5fk5c652lpnkw)
- [elasticsearch-pull-request #153690 / 9.4.3_bwc](https://gradle-enterprise.elastic.co/s/5nmmar36glt4y)
- [elasticsearch-pull-request #153664 / 8.19.17_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/lqtr4nsp52ot2)
- [elasticsearch-pull-request #153664 / 9.4.3_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/zqp2d2v66yfqw)
- [elasticsearch-pull-request #153664 / 8.19.17_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/or3jf5t7ptu2s)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.3.6#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/40_knn_search/Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn}" -Dtests.seed=F87A4B53C29E306B -Dtests.bwc=true -Dtests.locale=ne-IN -Dtests.timezone=US/Arizona -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dsearch.vectors%2F40_knn_search%2FVector%20rescoring%20has%20no%20effect%20for%20non-quantized%20vectors%20and%20provides%20same%20results%20as%20non-rescored%20knn%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: field [hits.hits.2._score] does not match the expected value
Expected: <1.0007358E-5>
     but: was <1.00073585E-5>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=search.vectors/40_knn_search/Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn}
- [main] 1 consecutive failures in step 9.3.6_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 6 failures in test test {p0=search.vectors/40_knn_search/Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn} (0.6% fail rate in 929 executions)
- [main] 2 failures in step 8.19.17_part_6_bwc-snapshots (0.9% fail rate in 216 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 234 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -617,12 +617,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
   method: test {p0=esql/160_union_types/load four indices with single conversion function TO_IP}
   issue: https://github.com/elastic/elasticsearch/issues/151413
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/40_knn_search/Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn}
-  issue: https://github.com/elastic/elasticsearch/issues/151418
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/42_knn_search_flat/Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn}
-  issue: https://github.com/elastic/elasticsearch/issues/151453
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
   method: test {feature:SUBQUERIES}
   issue: https://github.com/elastic/elasticsearch/issues/149681

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml
@@ -558,7 +558,7 @@ setup:
 "Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn":
   - requires:
       reason: 'Quantized vector rescoring is required'
-      test_runner_features: [capabilities]
+      test_runner_features: [capabilities, close_to]
       capabilities:
         - method: GET
           path: /_search
@@ -606,9 +606,9 @@ setup:
 
   # Compare scores as hit IDs may change depending on how things are distributed
   - match: { hits.total: 3 }
-  - match: { hits.hits.0._score: $knn_score0 }
-  - match: { hits.hits.1._score: $knn_score1 }
-  - match: { hits.hits.2._score: $knn_score2 }
+  - close_to: { hits.hits.0._score: { value: $knn_score0, error: 0.000001 }}
+  - close_to: { hits.hits.1._score: { value: $knn_score1, error: 0.000001 }}
+  - close_to: { hits.hits.2._score: { value: $knn_score2, error: 0.000001 }}
 
 ---
 "Dimensions are dynamically set":

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/42_knn_search_flat.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/42_knn_search_flat.yml
@@ -260,7 +260,7 @@ setup:
 "Vector rescoring has no effect for non-quantized vectors and provides same results as non-rescored knn":
   - requires:
       reason: 'Quantized vector rescoring is required'
-      test_runner_features: [capabilities]
+      test_runner_features: [capabilities, close_to]
       capabilities:
         - method: GET
           path: /_search
@@ -308,9 +308,9 @@ setup:
 
   # Compare scores as hit IDs may change depending on how things are distributed
   - match: { hits.total: 3 }
-  - match: { hits.hits.0._score: $knn_score0 }
-  - match: { hits.hits.1._score: $knn_score1 }
-  - match: { hits.hits.2._score: $knn_score2 }
+  - close_to: { hits.hits.0._score: { value: $knn_score0, error: 0.000001 }}
+  - close_to: { hits.hits.1._score: { value: $knn_score1, error: 0.000001 }}
+  - close_to: { hits.hits.2._score: { value: $knn_score2, error: 0.000001 }}
 ---
 "Test bad parameters":
   - do:

